### ISSUE TITLE
False positive for "Condition is duplicated in another branch"

### ISSUE THREAD

**Issue by Gonzalo1987** [kalessil/phpinspectionsea#1654]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.6 |
| Language level | e.g. PHP 7.4

Current behaviour
--

Hello, probably I'm wrong, but I believe that this inspection have no sense, what is wrong with this if? 

![imagen](https://user-images.githubusercontent.com/7010439/98113634-4077e080-1ea4-11eb-9585-dd6c753f23e0.png)

Thanks!

**Comment by Gonzalo1987** [comment]
Is easy to solve (with a probably suboptimal aproaching) so: 

![imagen](https://user-images.githubusercontent.com/7010439/98132302-850f7600-1ebc-11eb-808e-fdeed2f8eab0.png)

I'm saying that is probably suboptimal, because with the example the $b['t'] is only accessed once, (the same that with this aproaching) but is not stored in a variable, and since it is not used afterwards it is probably suboptimal. Probably this warning is for a better readability?

**Comment by BenceSzalai** [comment]
This happens because you are mapping out two independent decisions as if they were codependent. You must duplicate the conditions (either for `$a` or `$b['t']`) because you threat the result as an algebraic chain, but in fact examining `$a` and `$b['t']` has nothing to do with each other; you can check them one after the other.
```php
$e = '';
if ( !$a ) {
    $e .= '!';
}
$e .= 'a && ';
if ( !$b['t'] ) {
    $e .= '!';
}
$e .= 'b';
echo $e;
```
Or since you just echo them, even this way:
```php
if ( !$a ) {
    echo '!';
}
echo 'a && ';
if ( !$b['t'] ) {
    echo '!';
}
echo 'b';
```

Or if you want it more dense and kind of more readable in a linear sense of reading:
```php
echo ($a ? '' : '!') . 'a && ' . ($b ? '' : '!') . 'b';
```

This check also helps you to keep your cyclomatic complexity down. Sometimes the duplicated condition cannot be factored out, because of other more meaningful reasons, but in your case, you can write it with less complexity easily.

This inspection also sense when you have large branches and may not see all conditions at the first glance, and you may accidentally duplicate conditions which can cause some headaches, even thought it is also best to keep branches small and factor out huge chunks into methods/functions.

**Comment by ea-inspections-team** [comment]
Thank you for reporting @Gonzalo1987 and @BenceSzalai for joining the discussion!

A local variable holding the duplicated condition is the way to go. We updated the message to 'This condition is duplicated in another if/elseif branch (replacing duplicates with a local variable would make sense)'.

In this case, '$bt = $b['t'] === true' would make sense, but agree it looks redundant. We'll close the ticket and revisit this part of the inspection in future releases for making it more reasonable. Thanks for flagging!

**Comment by ea-inspections-team** [comment]
Update: checked on several biggest OSS projects and a discovered that the duplicates search is not bringing reasonable value. Dropped the duplicates search completely.
